### PR TITLE
Add more IBA cocktails

### LINKED ISSUE #5: Gomme is Simple Syrup
Gomme and simple syrup are the same thing.

### PR DESCRIPTION
**Pull request by wooorm**
Related-to: tmcw/flair#3.
Closes: tmcw/flair#5.

### CODE REVIEW

File: front/src/Data.elm [hunk 1]
```diff
@@ -13,12 +13,28 @@ type IngredientType
...
+    | Hurricane
+    | Margarita
+    -- | Sherry
+    -- | Snifter
+    | Wine
+    -- Other:
+    | CopperMug
+    | IrishCoffeeMug
+    | SteelCup
+    | ZombieGlass
```
Comment (wooorm) [inline front/src/Data.elm:45]: Used the naming from Wikipedia‘s glassware.

File: front/src/Data.elm [hunk 2]
```diff
@@ -130,7 +147,7 @@ angosturaBitters =
 
 peychaudsBitters : Material
 peychaudsBitters =
-    { name = "Peychaud's bitters", t = Bitters }
+    { name = "Peychaud’s bitters", t = Bitters }
```
Comment (wooorm) [inline front/src/Data.elm:158]: Better punctuation

File: front/src/Data.elm [hunk 3]
```diff
@@ -143,9 +160,9 @@ lemon =
     { name = "Lemon", t = Garnish }
 
 
-blackberries : Material
-blackberries =
-    { name = "Blackberries", t = Garnish }
+blackberry : Material
+blackberry =
+    { name = "Blackberry", t = Garnish }
```
Comment (wooorm) [inline front/src/Data.elm]: Singular seems better, throughout?

File: front/src/Data.elm [hunk 4]
```diff
@@ -245,17 +262,12 @@ raspberrySyrup =
 
 raspberryLiqueur : Material
 raspberryLiqueur =
-    { name = "Raspberry liqueur", t = Sweetener }
-
+    { name = "Raspberry liqueur", t = Spirit }
```
Comment (wooorm) [inline front/src/Data.elm]: I think a better typing makes sense. Spirit for 40+ percent, Wine for wine like stuff including vermouth. Liqueur for other things like this? 🤔

File: front/src/Data.elm [hunk 5]
```diff
@@ -245,17 +262,12 @@ raspberrySyrup =
...
-lemonPeel =
-    { name = "Lemon peel", t = Garnish }
 
-
-limeSlice : Material
-limeSlice =
-    { name = "Lime slice", t = Garnish }
+orange : Material
+orange =
+    { name = "Orange", t = Garnish }
```
Comment (wooorm) [inline front/src/Data.elm]: Instead opted to do peels, slices, wedges, rinds, etc in quantities. And use the whole ingredients here.

File: front/src/Data.elm [hunk 6]
```diff
@@ -280,14 +292,29 @@ cider =
 
 champagne : Material
 champagne =
-    { name = "Champagne", t = Spirit }
+    { name = "Champagne", t = Other }
```
Comment (wooorm) [inline front/src/Data.elm]: How broad to go? Are champagne, prosecco, all sparkling wines?

File: front/src/Data.elm [hunk 7]
```diff
@@ -373,6 +410,174 @@ lilletBlanc =
...
 
 
+irishWhiskey : Material
+irishWhiskey =
+    { name = "Irish whiskey", t = Spirit }
+
+
+oldTomGin : Material
+oldTomGin =
+    { name = "Old Tom Gin", t = Spirit }
```
Comment (wooorm) [inline front/src/Data.elm]: How, tom, *how* could you not use old tom, whilest you are called tom, and this website’s hostname includes *old*.

File: front/src/Main.elm [hunk 8]
```diff
@@ -472,6 +495,19 @@ drinkIcon recipe =
...
+        Hurricane ->
+            cocktailIcon
+        IrishCoffeeMug ->
+            cocktailIcon
+        SteelCup ->
+            cocktailIcon
+        Wine ->
+            cocktailIcon
+        ZombieGlass ->
+            cocktailIcon
```
Comment (wooorm) [inline front/src/Main.elm]: Where are icons from? How can I add them?

Comment (tmcw) [reply front/src/Main.elm]: So… they're in Icons.elm and I have been drawing them in Figma on a 50x50px frame, then running svgo to optimize them, and then rewriting the SVG in Elm's SVG 'vdom' syntax. A little crazy :)

File: front/src/Data.elm (hunk 1 above)
Comment (tmcw) [reply front/src/Data.elm:45]: TIL that an Irish cofee mug exists :)

File: front/src/Data.elm (hunk 4 above)
Comment (tmcw) [reply front/src/Data.elm]: Yeah, now that there are so many ingredients, more categories definitely make sense! Liqueur, wine, and maybe juice seem like good additional types?

File: front/src/Data.elm (hunk 5 above)
Comment (tmcw) [reply front/src/Data.elm]: Makes sense! If you have an orange you have an orange peel :)

File: front/src/Data.elm (hunk 7 above)
Comment (tmcw) [reply front/src/Data.elm]: Haha, I don't think I've even had old tom gin! Will definitely try it out once things settle down and I'll be older then too

File: front/src/Main.elm (hunk 8 above)
Comment (tmcw) [reply front/src/Main.elm]: (if you aren't set up with a SVG workflow you like, I can definitely draw these!)

Comment (wooorm) [reply front/src/Main.elm]: I could implement and optimize SVG but I can’t draw (wanting to get into that though, and been meaning to ask you about tips), so would appreciate if you could do that? Wine glass and hurricane are useful probably, margarita may be similar to standard cocktail. Zombie, steel, copper, irish coffee could maybe be grouped in one funky icon?

File: front/src/Main.elm [hunk 9]
```diff
@@ -41,7 +41,7 @@ type alias Model =
     , availableMaterials : MaterialSet
     , units : Units
     , mode : Mode
-    , substituteWhiskey : Bool
+    , pedantic : Bool
```
Comment (wooorm) [inline front/src/Main.elm:45]: Added a new pedantic mode, on by default, that does several substitutions when turned off (see below).

File: front/src/Main.elm [hunk 10]
```diff
@@ -127,15 +127,75 @@ fuzzyWhiskey substituteWhiskey model =
...
+fuzzyIngredient : Ingredient -> Ingredient
+fuzzyIngredient ingredient =
     { ingredient
         | material =
             if ingredient.material.name |> String.toLower |> String.contains "whiskey" then
                 { name = "Whiskey"
                 , t = Spirit
                 }
 
+            else if ingredient.material.name |> String.toLower |> String.contains "rum" then
```
Comment (wooorm) [inline front/src/Main.elm:140]: I don’t know elm so I didn’t know how to make this pretty 😭

File: front/src/Data.elm [hunk 11]
```diff
@@ -4,21 +4,45 @@ import Quantity exposing (Quantity(..))
 
 
 type IngredientType
```
Comment (wooorm) [inline front/src/Data.elm:6]: New grouping of ingredient types, do they make sense to you?

File: front/src/Main.elm (hunk 8 above)
Comment (tmcw) [reply front/src/Main.elm]: Okay, I've sketched up wine, hurricane, steel, and irish coffee glasses - I'll also do a margarita glass. I think at this teensy tiny scale, a zombie glass and a collins look so similar we should group 'em as I've been doing with highball & collins.

Comment (tmcw) [reply front/src/Main.elm]: Erg, I'm not sure how to commit to this PR - is the 'allow maintainer edits' checkbox checked? [Here's a branch locally with the icons!](https://github.com/tmcw/flair/tree/iba)

Comment (wooorm) [reply front/src/Main.elm]: Hmm yeah, it was allowed 🤷‍♂️ Screen Shot 2020-03-19 at 7 46 20 am

Comment (wooorm) [reply front/src/Main.elm]: Added your commit on top!

**Comment by wooorm** [comment]
Well, IBA published some more officials today, praise to RSS feeds. Added!

**Comment by tmcw** [comment]
Beautiful! As soon as you're happy with where this is at, I'll do whatever elm-flavored refactoring I can and get it merged

**Comment by wooorm** [comment]
I’m done! Ready for some elm-seasoning! 🍸

**Comment by wooorm** [comment]
And do check if you agree on the new ingredient groupings!

**Comment by tmcw** [comment]
There were enough new cocktails that all my O(n^2) code started acting up :) Did a bunch of refactoring so it's still zippy and it should be rebuilding on zeit momentarily!

**Comment by wooorm** [comment]
Awesome!

**Pull request merged**